addons/mail/static/src/discuss/core/common/notification_settings.js

```diff
@@ -0,0 +1,97 @@
+/* @odoo-module */
+
+import { useState, Component } from "@odoo/owl";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { _t } from "@web/core/l10n/translation";
+import { useService } from "@web/core/utils/hooks";
+
+export class NotificationSettings extends Component {
+    static components = { Dropdown, DropdownItem };
+    static props = ["hasSizeConstraints?", "thread", "close", "className?"];
+    static template = "discuss.NotificationSettings";
+
+    setup() {
+        this.threadService = useState(useService("mail.thread"));
+    }
+
+    get settingItems() {
+        return [
+            {
+                id: "all",
+                name: _t("All Messages"),
+            },
+            {
+                id: "mentions",
+                name: _t("Mentions Only"),
+            },
+            {
+                id: "none",
+                name: _t("Nothing"),
+            },
+        ];
+    }
+
+    get muteDurationItems() {
+        return [
+            {
+                id: "15_mins",
+                value: 15,
+                name: _t("For 15 minutes"),
+            },
+            {
+                id: "1_hour",
+                value: 60,
+                name: _t("For 1 hour"),
+            },
+            {
+                id: "3_hours",
+                value: 180,
+                name: _t("For 3 hours"),
+            },
+            {
+                id: "8_hours",
+                value: 480,
+                name: _t("For 8 hours"),
+            },
+            {
+                id: "24_hours",
+                value: 1440,
+                name: _t("For 24 hours"),
+            },
+            {
+                id: "forever",
+                value: -1,
+                name: _t("Until I turn it back on"),
+            },
+        ];
+    }
+
+    get muteUntilText() {
+        if (
+            this.props.thread.notificationMuteUntilDateTime &&
+            this.props.thread.notificationMuteUntilDateTime.year <= new Date().getFullYear() + 2
+        ) {
+            return (
+                _t("Until ") +
+                this.props.thread.notificationMuteUntilDateTime.toFormat("MM/dd, HH:mm")
+            );
+        }
+        // Forever is a special case, so we don't want to display the date.
+        return undefined;
+    }
+
+    selectUnmute() {
+        this.threadService.muteThread(this.props.thread);
+        this.props.close();
+    }
+
+    selectMuteDuration(item) {
+        this.threadService.muteThread(this.props.thread, { minutes: item.value });
```

Comment:
Extra param `minutes` would be better named `duration`, in both JS and route

Reply:
but in this case, what I wanted to specify is the unit of the duration rather than the "duration" word itself. So maybe keeping minute as the parameter is better? Maybe in the future, we will have `hours` `seconds` and then it is better to keep using `minutes`, which is more meaningful than duration